### ISSUE TITLE
Xml Deserializer Issue (Issue #147 cont.)

### ISSUE THREAD

**Issue by PT-BBalazs** [googleads/googleads-php-lib#154]
Dear Knack and Team,

we have managed to reproduce the issue mentioned in #147.

We generated the trace by modifying the xml deserializer php file in the following manner:

```
debug_print_backtrace();
   $deserializer = new XmlDeserializer(BatchJobUtils::$CLASS_MAP);
   $mutateResponse = $deserializer->ConvertXmlToObject($xmlResponse);
```

and the generated output is the following (sensitive information asterisked out):

```
#0 AddKeywordsUsingIncrementalBatchJob(AdWordsUser Object ([] => 10.0.0,[] => AwApi-PHP,[] => Prototype.co Online Services,[] => Array ([0] => https://www.googleapis.com/auth/adwords),[] => Array ([userAgent] => Prototype.co Online Services (AwApi-PHP/10.0.0, Common-PHP/10.0.0, PHP/5.4.45-0+deb7u2, BatchJobHelper),[clientCustomerId] => 6928223763,[developerToken] => 2EzLm0AluZTtDes_ciOGDQ),[] => https://adwords.google.com,[] => v201605,[] => /var/**************************************/www,[] => 1,[] => 1,[] => 0,[] => ,[] => ,[] => https://accounts.google.com,[] => Array ([client_id] => *******8265-**************bb7t1j.apps.googleusercontent.com,[client_secret] => **************I2Dhq9,[refresh_token] => 1/**************gLF7tVo,[access_token] => **************l93Hnc,[token_type] => Bearer,[expires_in] => 3600,[id_token] => **************,[timestamp] => 1470047112),[] => SimpleOAuth2Handler Object ([] => CurlUtils Object (),[] => https://accounts.google.com,[] => Array ([0] => https://www.googleapis.com/auth/adwords)),[] => true,[] => 10.0.0,[] => Common-PHP), Array ([0] => Array ([0] => 2,[1] => pos1,[2] => 18534000534,[3] => 243593454,[4] => best bar crawl budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => -0.50052765816284,[9] => 0.30910493671296,[10] => 0.71173954605586,[11] => -0.43980282012437,[12] => -0.5,[13] => -0.43980282012437),[1] => Array ([0] => 11,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => all night crash pubcrawl,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.57596052713612,[9] => -1.7884836528675,[10] => 0.59624343272677,[11] => -0.21493626390396,[12] => -0.14020914261503,[13] => -0.21493626390396),[2] => Array ([0] => 15,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => backpackers pub crawl budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.57596052713612,[9] => -3.5131016578519,[10] => 1.7021788321393,[11] => -0.22329388701435,[12] => -0.2596465603982,[13] => -0.22329388701435),[3] => Array ([0] => 20,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => budapest extreme pub crawl,[5] => Phrase,[6] => 10,[7] => 10,[8] => 0.57596052713612,[9] => 0.57359852338892,[10] => -1.6156273660983,[11] => -0.22329388701435,[12] => -0.2596465603982,[13] => -0.22329388701435),[4] => Array ([0] => 29,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => budapest ultimate pub crawl,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.57596052713612,[9] => 0.39769620930931,[10] => -0.50969196668579,[11] => -0.22329388701435,[12] => -0.2596465603982,[13] => -0.22329388701435),[5] => Array ([0] => 31,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => crawl pub budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.57596052713612,[9] => -1.3673841737072,[10] => -1.6156273660983,[11] => -0.22329388701435,[12] => -0.2596465603982,[13] => -0.22329388701435),[6] => Array ([0] => 34,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => original backpackers pubcrawl budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.57596052713612,[9] => -1.4831986449607,[10] => 1.7021788321393,[11] => -0.22329388701435,[12] => -0.2596465603982,[13] => -0.22329388701435),[7] => Array ([0] => 35,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => pub cralw budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.57596052713612,[9] => 1.967491103141,[10] => -0.50969196668579,[11] => -0.22329388701435,[12] => -0.2596465603982,[13] => -0.22329388701435),[8] => Array ([0] => 43,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => pub crwal budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.57596052713612,[9] => 0.66833379171554,[10] => -1.6156273660983,[11] => -0.22329388701435,[12] => -0.2596465603982,[13] => -0.22329388701435),[9] => Array ([0] => 50,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => ruins pub crawl budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.57596052713612,[9] => 0.63756300123605,[10] => -0.50969196668579,[11] => -0.22329388701435,[12] => -0.2596465603982,[13] => -0.22329388701435),[10] => Array ([0] => 51,[1] => pos1,[2] => 18534000414,[3] => 243593454,[4] => ruins pub crawl budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => -0.7025065317568,[9] => 0.58813590471781,[10] => 0.59624343272677,[11] => -0.21493626390396,[12] => -0.2596465603982,[13] => -0.21493626390396),[11] => Array ([0] => 56,[1] => pos1,[2] => 25712954094,[3] => 341580894,[4] => bar crawling budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.70656554816855,[9] => 0.96984707215461,[10] => -1.4773031863963,[11] => -0.63141675961899,[12] => -0.66815310478106,[13] => -0.63141675961899),[12] => Array ([0] => 58,[1] => pos1,[2] => 25712954094,[3] => 341580894,[4] => best bar crawl in budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.70656554816855,[9] => -1.4116637133329,[10] => 1.00787974399,[11] => -0.63141675961899,[12] => -0.66815310478106,[13] => -0.63141675961899),[13] => Array ([0] => 61,[1] => pos1,[2] => 25712954094,[3] => 341580894,[4] => budapest bars tour,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.70656554816855,[9] => 1.2387918250077,[10] => 1.00787974399,[11] => -0.63141675961899,[12] => -0.66815310478106,[13] => -0.63141675961899),[14] => Array ([0] => 62,[1] => pos1,[2] => 25712954094,[3] => 341580894,[4] => budapestbar crawl,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.70656554816855,[9] => 1.2392917223179,[10] => -1.4773031863963,[11] => -0.63141675961899,[12] => -0.66815310478106,[13] => -0.63141675961899),[15] => Array ([0] => 63,[1] => pos1,[2] => 25712954094,[3] => 341580894,[4] => guided bar tours budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.70656554816855,[9] => 0.78388527278404,[10] => 1.00787974399,[11] => -0.63141675961899,[12] => -0.66815310478106,[13] => -0.63141675961899),[16] => Array ([0] => 66,[1] => pos1,[2] => 25712953974,[3] => 341580894,[4] => backpackers pub crawl budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.55215046005257,[9] => -0.95024836601276,[10] => -0.18626714965571,[11] => -0.23590119781483,[12] => -0.29754643161085,[13] => -0.23590119781483),[17] => Array ([0] => 67,[1] => pos1,[2] => 25712953974,[3] => 341580894,[4] => best budapest pub crawl,[5] => Phrase,[6] => 00,[7] => 10,[8] => -0.84630067094702,[9] => -2.008364795985,[10] => -0.80715764850809,[11] => -0.22341965295691,[12] => -0.29754643161085,[13] => -0.22341965295691),[18] => Array ([0] => 81,[1] => pos1,[2] => 25712953974,[3] => 341580894,[4] => pub brawl budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.55215046005257,[9] => 0.12803492211383,[10] => -0.18626714965571,[11] => -0.23590119781483,[12] => -0.29754643161085,[13] => -0.23590119781483),[19] => Array ([0] => 85,[1] => pos1,[2] => 25712953974,[3] => 341580894,[4] => pub crawl budapest bachelor,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.55215046005257,[9] => 0.1915390711957,[10] => -0.18626714965571,[11] => -0.23590119781483,[12] => -0.29754643161085,[13] => -0.23590119781483),[20] => Array ([0] => 99,[1] => pos1,[2] => 25712953974,[3] => 341580894,[4] => puscrawl budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.55215046005257,[9] => 0.70729574144169,[10] => -1.4280481473605,[11] => -0.23590119781483,[12] => -0.29754643161085,[13] => -0.23590119781483),[21] => Array ([0] => 101,[1] => pos1,[2] => 25712953974,[3] => 341580894,[4] => ruin pubs tour budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.55215046005257,[9] => 0.21084776517329,[10] => 1.055513848049,[11] => -0.23590119781483,[12] => -0.29754643161085,[13] => -0.23590119781483),[22] => Array ([0] => 102,[1] => pos1,[2] => 25712953974,[3] => 341580894,[4] => the original budapest pub crawl,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.55215046005257,[9] => 0.78667592779402,[10] => -1.4280481473605,[11] => -0.23590119781483,[12] => -0.29754643161085,[13] => -0.23590119781483),[23] => Array ([0] => 109,[1] => pos1,[2] => 19527570294,[3] => 255213534,[4] => beste bar in budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 2.5408846536335,[9] => 0.98812761853665,[10] => 0.088259156328138,[11] => -0.83822639033724,[12] => -0.83602340898318,[13] => -0.83822639033724),[24] => Array ([0] => 110,[1] => pos1,[2] => 19527570294,[3] => 255213534,[4] => beste bars in budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 0.84417140052827,[9] => -1.6107797672641,[10] => 1.0811746650197,[11] => -0.82550457328322,[12] => -0.83602340898318,[13] => -0.82550457328322),[25] => Array ([0] => 118,[1] => pos3,[2] => 19527570414,[3] => 255213534,[4] => map,[5] => Phrase,[6] => 00,[7] => 01,[8] => 1.579542844149,[9] => 0.85151153862842,[10] => -1.2472191289246,[11] => -0.63573077254904,[12] => -0.63020879124888,[13] => -0.63573077254904),[26] => Array ([0] => 119,[1] => pos1,[2] => 19527570414,[3] => 255213534,[4] => budapest ruins bar,[5] => Phrase,[6] => 00,[7] => 10,[8] => 1.579542844149,[9] => 0.24354475195771,[10] => -1.2472191289246,[11] => -0.63573077254904,[12] => -0.63020879124888,[13] => -0.63573077254904),[27] => Array ([0] => 122,[1] => pos1,[2] => 19527570414,[3] => 255213534,[4] => ruins bar budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => -0.68786381248305,[9] => 0.77961829393214,[10] => 0.93541434669349,[11] => -0.44796218993962,[12] => -0.63020879124888,[13] => -0.44796218993962),[28] => Array ([0] => 125,[1] => pos1,[2] => 19527570534,[3] => 255213534,[4] => budapest ruin pub,[5] => Phrase,[6] => 00,[7] => 10,[8] => -0.38144461635518,[9] => 0.44697696794724,[10] => -1.0214463280839,[11] => -0.54060866389374,[12] => -0.50469493868284,[13] => -0.54060866389374),[29] => Array ([0] => 127,[1] => pos1,[2] => 19527570534,[3] => 255213534,[4] => ruin pub budapest,[5] => Phrase,[6] => 10,[7] => 10,[8] => -0.65335513645424,[9] => 0.16492490015145,[10] => 1.6128099917114,[11] => 0.28246802688448,[12] => 1.4580076006393,[13] => 0.28246802688448),[30] => Array ([0] => 130,[1] => pos1,[2] => 19527570534,[3] => 255213534,[4] => ruinpubs budapest,[5] => Phrase,[6] => 00,[7] => 10,[8] => 2.0002220788368,[9] => 2.1779165889062,[10] => -0.64512399668456,[11] => -0.69197908978399,[12] => -0.89723544654727,[13] => -0.69197908978399))) called at [/var/**************************************/www/wp-content/plugins/guppi/Functions.php:618] 
#1 guppi_addkeywords() called at [/var/**************************************/www/wp-content/plugins/guppi/guppi.php:249] 
#2 guppi_adwords_table(, , guppi_plugin_adwords_table) 
#3 call_user_func(guppi_adwords_table, , , guppi_plugin_adwords_table) called at [/var/**************************************/www/wp-includes/shortcodes.php:326] 
#4 do_shortcode_tag(Array ([0] => [guppi_plugin_adwords_table],[1] => ,[2] => guppi_plugin_adwords_table,[3] => ,[4] => ,[5] => ,[6] => )) 
#5 preg_replace_callback(/\[(\[?)(guppi_plugin_adwords_table)(?![\w-])([^\]\/]*(?:\/(?!\])[^\]\/]*)*?)(?:(\/)\]|\](?:([^\[]*+(?:\[(?!\/\2\])[^\[]*+)*+)\[\/\2\])?)(\]?)/, do_shortcode_tag, [guppi_plugin_adwords_table] ) called at [/var/**************************************/www/wp-includes/shortcodes.php:223] 
#6 do_shortcode([guppi_plugin_adwords_table] ) 
#7 call_user_func_array(do_shortcode, Array ([0] => [guppi_plugin_adwords_table] )) called at [/var/**************************************/www/wp-includes/plugin.php:235] 
#8 apply_filters(the_content, [guppi_plugin_adwords_table]) called at [/var/**************************************/www/wp-includes/post-template.php:240] 
#9 the_content() called at [/var/**************************************/www/wp-content/themes/prototype/index.php:6] 
#10 include(/var/**************************************/www/wp-content/themes/prototype/index.php) called at [/var/**************************************/www/wp-includes/template-loader.php:75] 
#11 require_once(/var/**************************************/www/wp-includes/template-loader.php) called at [/var/**************************************/www/wp-blog-header.php:19] 
#12 require(/var/**************************************/www/wp-blog-header.php) called at [/var/**************************************/www/index.php:17]
Fatal error: Call to a member function setAccessible() on a non-object in /var/**************************************/www/wp-content/plugins/guppi/googleads-php-lib-master/src/Google/Api/Ads/AdWords/Util/XmlDeserializer.php on line 134

Array
(
    [type] => 1
    [message] => Call to a member function setAccessible() on a non-object
    [file] => /var/**************************************/www/wp-content/plugins/guppi/googleads-php-lib-master/src/Google/Api/Ads/AdWords/Util/XmlDeserializer.php
    [line] => 134
)
```

Please keep us informed on the issue and lend us any insight on what might be the problem.

Thanks for your help.

Edit: Also, here's the content of $xmlResponse to help you out:

`18534000534125376368454KEYWORDKeywordbest bar crawl budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDAPPROVEDMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP018534000414209417357814KEYWORDKeywordall night crash pubcrawlPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP118534000414209417358054KEYWORDKeywordbackpackers pub crawl budapestPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP218534000414209417359014KEYWORDKeywordbudapest extreme pub crawlPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP318534000414139563865921KEYWORDKeywordbudapest ultimate pub crawlPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP418534000414203445869574KEYWORDKeywordcrawl pub budapestPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP518534000414209417359734KEYWORDKeywordoriginal backpackers pubcrawl budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP618534000414209417360214KEYWORDKeywordpub cralw budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP718534000414141043373814KEYWORDKeywordpub crwal budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDAPPROVEDMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP818534000414209417361174KEYWORDKeywordruins pub crawl budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP918534000414209417361174KEYWORDKeywordruins pub crawl budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP1025712954094125376367494KEYWORDKeywordbar crawling budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDAPPROVEDMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP1125712954094125376368694KEYWORDKeywordbest bar crawl in budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDAPPROVEDMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP1225712954094209417358774KEYWORDKeywordbudapest bars tourPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP1325712954094209417359254KEYWORDKeywordbudapestbar crawlPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP1425712954094209417359494KEYWORDKeywordguided bar tours budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP1525712953974209417358054KEYWORDKeywordbackpackers pub crawl budapestPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP1625712953974180965671482KEYWORDKeywordbest budapest pub crawlPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP17operations[18].operand.criterion.textbrawlCriterionPolicyError.POLICY_ERRORCriterionPolicyErrorviolencebrawlGoogle AdWordsThis ad or keyword requires review before it can run.true451825712953974209417360454KEYWORDKeywordpub crawl budapest bachelorPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP1925712953974209417360694KEYWORDKeywordpuscrawl budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP2025712953974209417360934KEYWORDKeywordruin pubs tour budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP2125712953974209417361414KEYWORDKeywordthe original budapest pub crawlPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWMANUAL_CPCCAMPAIGNManualCpcBiddingSchemetrueCpcBidMoney200000000ADGROUPCpmBidMoney50000000ADGROUP2219527570294209417358294KEYWORDKeywordbeste bar in budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWBUDGET_OPTIMIZERCAMPAIGNBudgetOptimizerBiddingSchemeMoney0falseCpcBidMoney68780000ADGROUPCpmBidMoney50000000ADGROUP2319527570294209417358534KEYWORDKeywordbeste bars in budapestPHRASEBiddableAdGroupCriterionENABLEDRARELY_SERVEDPENDING_REVIEWBUDGET_OPTIMIZERCAMPAIGNBudgetOptimizerBiddingSchemeMoney0falseCpcBidMoney68780000ADGROUPCpmBidMoney50000000ADGROUP24255213534true11010323KEYWORDKeywordmapPHRASENegativeCampaignCriterion2519527570414209413292214KEYWORDKeywordbudapest ruins barPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWBUDGET_OPTIMIZERCAMPAIGNBudgetOptimizerBiddingSchemeMoney0falseCpcBidMoney68780000ADGROUPCpmBidMoney50000000ADGROUP2619527570414185261684969KEYWORDKeywordruins bar budapestPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWBUDGET_OPTIMIZERCAMPAIGNBudgetOptimizerBiddingSchemeMoney0falseCpcBidMoney68780000ADGROUPCpmBidMoney50000000ADGROUP2719527570534209413291974KEYWORDKeywordbudapest ruin pubPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWBUDGET_OPTIMIZERCAMPAIGNBudgetOptimizerBiddingSchemeMoney0falseCpcBidMoney68780000ADGROUPCpmBidMoney50000000ADGROUP281952757053458424473673KEYWORDKeywordruin pub budapestPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWBUDGET_OPTIMIZERCAMPAIGNBudgetOptimizerBiddingSchemeMoney0falseCpcBidMoney68780000ADGROUPCpmBidMoney50000000ADGROUP2919527570534209413292454KEYWORDKeywordruinpubs budapestPHRASEBiddableAdGroupCriterionENABLEDELIGIBLEPENDING_REVIEWBUDGET_OPTIMIZERCAMPAIGNBudgetOptimizerBiddingSchemeMoney0falseCpcBidMoney68780000ADGROUPCpmBidMoney50000000ADGROUP30`

**Comment by fiboknacky** [comment]
Dear @PT-BBalazs,

Could you please tell me also what were the operations you've uploaded?
It seems you've modified our **AddKeywordsUsingIncrementalBatchJob** example.
How did you do that?

Thanks so much for your information, but it would be faster if I could try running that script by myself.
The `$xmlResponse` you gave me seems to be flattened out so I can figure out what's wrong in there.

Is it possible for you to trace back to the operations you've uploaded?

Best,
Knack

**Comment by PT-BBalazs** [comment]
Dear Knack,

We can't trace it back exactly, because our logs are generated based on the information returned at the end of the process. I think we can provide you the code though, and we will also be doing further work this time of the month, so the error is bound to pop up again.

I can't help but notice though that the xmlResponse is only partial: its end has been cut off, just barely into the AdGroupID(?) as you can see. Could that be the direct cause of the issue? I mean, the fatal error claims that setAccessible() has failed - I haven't checked, but if that function has anything to do with AdGroups, you might have a lead there.

Also, whenever we receive a fatal error, it is going to be this one. So if setAccessible() has anything to do with AdGroups, something tends to fail with those Ad Group IDs. We've been receiving this error for various clients, so this appears to be more of a general than a specific issue.

Thanks,
Balazs

**Comment by fiboknacky** [comment]
Dear @PT-BBalazs 

Thanks for more information.
I don't have enough time to look at this deeply yet, but will have a look soon.

Thanks so much for reporting this.

Knack

**Comment by PT-BBalazs** [comment]
Dear Knack,

Here is some extra bit of information. We have received this error once again. Looking back at the keywords we wanted to upload, we noticed that there was a keyword with an '&' character inside. Trying for blind luck, we removed that item and added a different one. The job uploaded fine just afterwards.

Maybe you could try catching this error somehow and return an error message to keep people informed. You might also want to check for other special characters as well.

Thanks,
Balazs

**Comment by fiboknacky** [comment]
Dear Balazs,

This is really helpful.
Actually special characters were encoded using `htmlspecialchars` already in [XmlSerializer](https://github.com/googleads/googleads-php-lib/blob/master/src/Google/Api/Ads/AdWords/Util/XmlSerializer.php#L128).
Let me check on my side and confirm what's the real cause of this.

Thanks so much!
Cheers,
Knack

**Comment by fiboknacky** [comment]
Dear Balazs,

I already double-checked and as I mentioned in the previous comment, & is already taken care of by `htmlspecialchars` so it shouldn't have effect on this.

Looking back to your comment above:
> I can't help but notice though that the xmlResponse is only partial: its end has been cut off, just barely into the AdGroupID(?) as you can see. Could that be the direct cause of the issue? I mean, the fatal error claims that setAccessible() has failed - I haven't checked, but if that function has anything to do with AdGroups, you might have a lead there.
It seems quite weird that xmlResponse you get back is partial. It looks more like errors on BatchJobService side and thus fails our deserialization process.

I know that the errors are not often, but if you face with this issue again, could you please pass me the batch job ID?
Thanks so much in advance.

Cheers,
Knack

**Comment by PT-BBalazs** [comment]
Dear Knack,
Thanks for the info, I will get back to you with an ID once we get an error. Indeed errors are few and far between, but they are reproducable (i.e. if you try to upload the very same set of keywords again). I'll keep you informed.

**Comment by fiboknacky** [comment]
Hi Balazs,

Really appreciate your help. :)
By the way, if your set of keywords is not so large, could you please pass them to me?
If you're concerned about privacy, I can give you google email to send instead.

Thanks in advance.

Best,
Knack

**Comment by PT-BBalazs** [comment]
Hi Knack,

We have just encountered this error after trying to upload a 3-hours long work session. Quite frustrating, really. The BatchJobId is 364145122, but the keyword set is unfortunately quite massive.

Actually, we are testing with live data, so privacy is indeed a concern. If you could provde us with a google mail address that would be excellent.

Thanks,
Balazs

**Comment by fiboknacky** [comment]
Hi Balazs,

Could you please send an email with your batch job ID and clientCustomerID to adwordsapi-support@google.com?
Please name the title accordingly so the person who opens that email first know that he/she should assign the case to me.

Thanks so much!
Knack

**Comment by PT-BBalazs** [comment]
Dear Knack,

Here's another one, resulting in the same fatal error:

BATCHJOBID
374799180

CLIENTCUSTOMERID
2927880327

In a minute, I will send an e-mail with some logs attached. The e-mail issue ID is [3-4144000013524].

Thanks for your help.

**Comment by fiboknacky** [comment]
Per personal communication, the issue seems to be affected by the old version of `XmlDeserializer` from v10.0.0. 
This has been fixed in [v12.0.0](https://github.com/googleads/googleads-php-lib/releases/tag/12.0.0), so upgrading to this version should fix this.
